use-shopping-cart/src/index.test.js

```diff
@@ -171,6 +171,45 @@ describe('useShoppingCart()', () => {
       expect(cart.current.cartCount).toBe(2)
       expect(cart.current.totalPrice).toBe(650)
     })
+
+    it('adds price_data from the metadata object from the 3rd param', () => {
+      const product = mockProduct()
+
+      act(() => {
+        cart.current.addItem(product, 1, {
+          type: 'tacos',
+          test: 'testing'
+        })
+      })
+
+      const entry = cart.current.cartDetails[product.id]
+
+      entry.price_data = {
+        type: 'tacos',
+        test: 'testing'
+      }
+
+      expect(cart.current.cartDetails).toStrictEqual({ [entry.id]: entry })
+    })
+
+    it('successfully stacks data to price_data if there is already content there', () => {
+      const product = mockProduct({ price_data: { test: 'static metadata' } })
+
+      act(() => {
+        cart.current.addItem(product, 1, {
+          dynamicTest: 'dynamic data'
+        })
+      })
+
+      const entry = cart.current.cartDetails[product.id]
+
+      entry.price_data = {
+        test: 'static metadata',
+        dynamicTest: 'dynamicData'
+      }
+
+      expect(cart.current.cartDetails).toStrictEqual({ [entry.id]: entry })
+    })
```

Comment:
Same thing here as above. I would also suggest perhaps adding a test for the `product_metadata` or adding it to the tests because right now there only tests for `price_metadata`.